arlas-core/src/main/java/io/arlas/server/utils/ParamsParser.java

```diff
@@ -136,52 +141,39 @@ private static HitsFetcher getHitsFetcher(String fetchHitsString) throws ArlasEx
-                    } else {
-                        throw new InvalidParameterException(CheckParams.INVALID_FETCHGEOMETRY);
-                    }
-                } else if (fetchOptions.length == 3) {
-                    String field = fetchOptions[1];
-                    // TODO check field existence ?
-                    String option = fetchOptions[2];
-                    if (option.equals(AggregatedGeometryStrategyEnum.first.name())) {
-                        aggregatedGeometry = new AggregatedGeometry(AggregatedGeometryStrategyEnum.first, field);
-                    } else if (option.equals(AggregatedGeometryStrategyEnum.last.name())) {
-                        aggregatedGeometry = new AggregatedGeometry(AggregatedGeometryStrategyEnum.last, field);
-                    } else {
-                        throw new InvalidParameterException(CheckParams.INVALID_FETCHGEOMETRY);
-                    }
-                } else {
-                    throw new InvalidParameterException(CheckParams.INVALID_FETCHGEOMETRY);
-                }
+    private static RawGeometries getAggregationRawGeometries(String rawGeometriesString, CollectionReference collectionReference) throws ArlasException {
+        RawGeometries rawGeometries = null;
+        if (!StringUtil.isNullOrEmpty(rawGeometriesString)) {
+            Matcher sortMatcher = Pattern.compile("\\((.*?)\\)$").matcher(rawGeometriesString);
+            rawGeometries = new RawGeometries();
+            if (sortMatcher.find()) {
+                rawGeometries.sort = sortMatcher.group(1);
             } else {
-                if (fetchGeometryString.equals("")) {
-                    aggregatedGeometry = new AggregatedGeometry(AggregatedGeometryStrategyEnum.byDefault);
-                } else {
-                    throw new InvalidParameterException(CheckParams.INVALID_FETCHGEOMETRY);
-                }
+                rawGeometries.sort = collectionReference.params.timestampPath;
+            }
+            String geometriesString = sortMatcher.replaceAll("");
+            List<String> geometries = Arrays.asList(geometriesString.split(",")).stream().distinct().collect(Collectors.toList());
+            if (geometries.isEmpty()) {
+                throw new InvalidParameterException(INVALID_AGG_RETURNED_GEOMETRIES);
+            } else {
+                rawGeometries.geometries = geometries;
             }
+        }
+        return rawGeometries;
+    }
 
+    private static List<AggregatedGeometryEnum> getAggregatedGeometries(String aggregatedGeometriesString) throws ArlasException {
+        List<AggregatedGeometryEnum> aggregatedGeometries = null;
+        if (!StringUtil.isNullOrEmpty(aggregatedGeometriesString)) {
+            List<String> aggregatedGeometriesStrings = Arrays.asList(aggregatedGeometriesString.split(",")).stream().distinct().collect(Collectors.toList());
+            for (String ag : aggregatedGeometriesStrings) {
```

Comment:
you could also use a Set to avoid the `distinct()`

Reply:
fixed